### ISSUE TITLE
Escape to end emulation

### ISSUE THREAD

**Issue by tony971** [libretro/RetroArch#3006]
I'd like to be able to contextually have the escape key end emulation and return to the main menu if a game is running. Hitting escape from the menu should still cause retroarch to close.

**Comment by RobLoach** [comment]
By default, Menu Toggle in the menu will open the menu (`F1`), Quit Retroarch will Quit (`Esc`). Are you suggesting allowing use of the same button to do the same thing based on whether the menu is open?

1. Playing game...
2. Press Escape to open the menu
3. Now that menu is open, press Escape again to quit

**Comment by tony971** [comment]
Right. I'd like to have them use the same key, alternating uses depending on whether the game is currently running or not.

**Comment by fr500** [comment]
Hmmm I don't like the idea, this would be catering to your particular needs and making a special case for everyone at the beginning of retroarch has lead to many problems. 

There are many people that don't want to go thru the intermediate step and many are ok with different keys for each action, I'd vote no.

**Comment by tony971** [comment]
I wasn't hoping for it to be mandatory, just an option. Maybe even a config-only option.

**Comment by fr500** [comment]
Even then it would be a hack on top of several other hacks.
Menu input is not really flexible. Anyway I just said I vote no, that's just me.

If someone is willing to implement and @twinaphex wants to merge it would be alright.
